docs/server/interceptors.md

```diff
@@ -0,0 +1,301 @@
+`On{{BeforeOperation}/{AfterOperation}}{TargetName}`.
+
+<table style="width: 100%;">
+    <tr>
+        <td>The possible values for <code>{BeforeOperation}</code> are:</td>
+        <td>The possible values for <code>{AfterOperation}</code> are:</td>
+        <td>The possible values for <code>{TargetName}</code> are:</td>
+    </tr>
+    <tr>
+        <td>
+            <ul style="margin-bottom: 0;">
+                <li>Inserting</li>
+                <li>Updating</li>
+                <li>Deleting</li>
+                <li>Executing</li>
+            </ul>
+        </td>
+        <td>
+            <ul style="margin-bottom: 0;">
+                <li>Inserted</li>
+                <li>Updated</li>
+                <li>Deleted</li>
+                <li>Executed</li>
+            </ul>
+        </td>
+        <td style="vertical-align: text-top;">
+            <ul style="margin-bottom: 0;">
+                <li><i>EntitySetName</i></li>
+                <li><i>ActionName</i></li>
+            </ul>
+        </td>
+    </tr>
+</table>
+
+### Example
+
+The example below demonstrates how both types of `{TargetName}` can be used. 
+
+- The first method shows a simple way to prevent *any*  user from deleting a particular EntitySet.
+- The second method shows how you can integrate role-based security using multiple techniques.
+- The third method shows how to prevent execution a custom Action.
+
+```cs
+using Microsoft.Restier.Providers.EntityFramework;
+using System;
+using System.Security.Claims;
+
+namespace Microsoft.OData.Service.Sample.Trippin.Api
+{
+
+    ///<summary>
+    /// Customizations to the EntityFrameworkApi for the TripPin service.
+    ///</summary>
+    ///<example>
+    /// Add the following line in WebApiConfig.cs to register this code:
+    /// await config.MapRestierRoute<TrippinApi>("Trippin", "api", new RestierBatchHandler(GlobalConfiguration.DefaultServer));
+    ///</example>
+    public class TrippinApi : EntityFrameworkApi<TrippinModel>
+    {
+        
+        ///<summary>
+        /// Specifies whether or not a Trip can be deleted from an EntitySet.
+        ///</summary>
+        protected void OnInsertingTrip(Trip trip)
+        {
+            Trace.WriteLine($"{DateTime.Now.ToString()}: {trip.TripId} is being Inserted.");
+            
+            if (string.IsNullOrWhiteSpace(trip.Description))
+            {
+                throw new ODataException("The Trip Description cannot be blank.");
+            }
+        }
+
+        ///<summary>
+        /// Specifies whether or not a Trip can be deleted from an EntitySet.
+        ///</summary>
+        protected void OnInsertedTrip(Trip trip)
+        {
+            Trace.WriteLine($"{DateTime.Now.ToString()}: {trip.tripId} has been Inserted.");
+
+            // Pseudocode that represents a real business process.
+            // EmailManager.SendTripWelcome(trip);
+        }
+
+    }
+
+}
+```
+
+## Centralized Interception
+
```

Comment:
This one we consider as authorization, also we are creating some end to end examples on security. Want to separate to security section.

And we are creating end to end sample on security, refer to commit https://github.com/chinadragon0515/RESTier/commit/08ae85b14d65eee56f36ad7dac8f035a4770add3 to have a preview.
Then we will remove an out of data RESTier model (not part of 0.5 release)